client/state/data-layer/wpcom/sites/admin-menu/index.js

```diff
@@ -20,7 +20,11 @@ export const requestFetchAdminMenu = ( action ) =>
 
 const sanitizeUrl = ( url, wpAdminUrl ) => {
 	const isSafeInternalUrl = new RegExp( '^/' ).test( url );
-	const isSafeWpAdminUrl = new RegExp( `^${ wpAdminUrl }` ).test( url );
+	// The replace function removes the protocol.
+	const isSafeWpAdminUrl = new RegExp( `^${ wpAdminUrl?.replace( /(^\w+:|^)\/\//, '' ) }` ).test(
+		url?.replace( /(^\w+:|^)\/\//, '' )
+	);
```

Comment:
What do you think about using this more readable regex?

```suggestion
	const isSafeWpAdminUrl = new RegExp( `^${ wpAdminUrl?.replace( /^https?:\/\//, '' ) }` ).test(
		url?.replace( /^https?:\/\//, '' )
	);
```

Reply:
I guess it's ok since we only expect an http protocol.